File: cpp/base/src/main/vsdk/toolkit/gui/feedback/ProgressMonitorInRam.cpp
#include "vsdk/toolkit/common/VSDK.h"
#include "vsdk/toolkit/gui/feedback/ProgressMonitorInRam.h"
ProgressMonitorInRam::ProgressMonitorInRam()
    : currentPercent(0)
{
}

void ProgressMonitorInRam::begin()
{
    currentPercent = 0;
}

void ProgressMonitorInRam::end()
{
    currentPercent = 100.0;
}

void ProgressMonitorInRam::update(double minValue, double maxValue, double currentValue)
{
    if ( (maxValue - minValue) < VSDK::EPSILON ) {
        return;
    }
    currentPercent = 100 * (currentValue - minValue) / (maxValue - minValue);
}

double ProgressMonitorInRam::getCurrentPercent()
{
    return currentPercent;
}

File: typescript/base/src/main/vsdk/toolkit/gui/feedback/ProgressMonitorInRam.ts
import { VSDK } from "../../common/VSDK.js";
import { ProgressMonitor } from "./ProgressMonitor.js";

/** In-memory progress monitor for UI bindings and polling clients. */
export class ProgressMonitorInRam extends ProgressMonitor {
  private currentPercent = 0;
  public begin(): void { this.currentPercent = 0; }
  public end(): void { this.currentPercent = 100; }
  public update(minValue: number, maxValue: number, currentValue: number): void { if (maxValue - minValue < VSDK.EPSILON) return; this.currentPercent = 100 * (currentValue - minValue) / (maxValue - minValue); }
  public getCurrentPercent(): number { return this.currentPercent; }
}
